docs/fastdds/ros2/Discovery_service/ros2_discovery_service.rst

```diff
@@ -0,0 +1,337 @@
+.. ros2-discovery-service
+
+Use ROS2 with Fast-DDS Discovery Service
+========================================
+
+In this section we are going to explain how to run some ROS2 examples using the Discovery Services
+as discovery communication.
+In order to get more information about the specific use of this configuration,
+please check the :ref:`Discovery Service Documentation <discovery_service>`
+or read the :ref:`common use cases <discovery-service-use-case>` for this configuration.
+
+The following tutorial gathers the steps to check this functionality and learn how to use it with ROS2.
+
+.. contents::
+    :local:
+    :backlinks: none
+    :depth: 2
```

Comment:
I think it's weird to have the index in the middle of the intro

Reply:
This design was chosen following ROS2 tutorial designs. As the use of internal-page indexes is rare along the documentation, it does not seem as a bad option for me.